nf_core/modules/update.py

```diff
@@ -0,0 +1,231 @@
+
+        Args:
+            module: If not None, will update only this module
+            all_modules: if True, all modules will be updated
+        """
+
+        # Get a list of all installed nf-core modules
+        nfcore_modules = self.get_installed_modules()
+
+        # Load the modules.json file
+        try:
+            with open(os.path.join(self.dir, "modules.json"), "r") as fh:
+                self.modules_json = json.load(fh)
+        except FileNotFoundError:
+            log.error("Could not load 'modules.json' file!")
+            sys.exit(1)
+
+        # Prompt for module or all
+        if module is None and not all_modules:
+            question = {
+                "type": "list",
+                "name": "all_modules",
+                "message": "Update all modules or a single named module?",
+                "choices": ["All modules", "Named module"],
+            }
+            answer = questionary.unsafe_prompt([question], style=nf_core.utils.nfcore_question_style)
+            if answer["all_modules"] == "All modules":
+                self.all_modules = True
+            else:
+                module = questionary.autocomplete(
+                    "Tool name:",
+                    choices=[m.module_name for m in nfcore_modules],
+                    style=nf_core.utils.nfcore_question_style,
+                ).ask()
+
+        # Only update the given module
+        if module:
+            if self.all_modules:
+                raise ModuleUpdateException("You cannot specify a tool and request all tools to be updated.")
+            nfcore_modules = [m for m in nfcore_modules if m.module_name == module]
+            if len(nfcore_modules) == 0:
+                raise ModuleUpdateException(f"Could not find the specified module: '{module}'")
+
+        # Update all modules
+        if len(nfcore_modules) > 0:
+            progress_bar = rich.progress.Progress(
+                "[bold blue]{task.description}",
+                rich.progress.BarColumn(bar_width=None),
+                "[magenta]{task.completed} of {task.total}[reset] » [bold yellow]{task.fields[test_name]}",
+                transient=True,
+            )
+        with progress_bar:
+            update_progress = progress_bar.add_task(
+                "Updating nf-core modules", total=len(nfcore_modules), test_name=nfcore_modules[0].module_name
+            )
+            for mod in nfcore_modules:
+                progress_bar.update(update_progress, advance=1, test_name=mod.module_name)
+                if self.update_module(mod):
+                    self.updated.append(mod)
+                else:
+                    self.up_to_date.append(mod)
+
+        # Print the results
+        self._print_results()
+
+    def update_module(self, mod):
+        """
+        Update a single module
+        """
+        # Compare content to remote
+        files_to_check = ["main.nf", "functions.nf", "meta.yml"]
```

Comment:
I've been trying out the `update` command locally, and noticed that all modules don't have a meta.yml file. Hence the update command fails when it tries to download it from the remote.

Reply:
Ah very nice, thanks for finding that! Will build around this 👍